config.go

```diff
@@ -84,6 +84,7 @@ func ConfigStringToClusterConfig(configString string) (*gocql.ClusterConfig, err
 	}
 
 	passwordAuthenticator := gocql.PasswordAuthenticator{}
+	var sslOptions *gocql.SslOptions
```

Comment:
What happens if sslOptions is set to `&gocql.SslOptions{}` and all the if `sslOptions == nil {` are removed? Is an empty `&gocql.SslOptions{}` the same as a nil one?

If it is the same, then can just set `&gocql.SslOptions{}` in clusterConfig and remove `var sslOptions`

Reply:
[they are treated the same](https://github.com/gocql/gocql/blob/95d072f1b5bbc604ce10f41ce1c914578592ed00/connectionpool.go#L30-L33). is it robust to rely on that?

Reply:
Ok then, but is var sslOptions needed? Can clusterConfig.SslOpts be used directly?

Reply:
> they are treated the same. is it robust to rely on that?

oops i'd referenced the wrong code last night. it's actually [here](https://github.com/gocql/gocql/blob/95d072f1b5bbc604ce10f41ce1c914578592ed00/connectionpool.go#L85-L86)

``` go
if cfg.SslOpts != nil {
    tlsConfig, err = setupTLSConfig(cfg.SslOpts)
```

so actually an empty `&gocql.SslOptions{}` isn't treated the same as a `nil` one

---

> Ok then, but is var sslOptions needed? Can clusterConfig.SslOpts be used directly?

we could use `clusterConfig.SslOpts` directly with `if ...` or I think we should follow how you approached `clusterConfig.Authenticator`

``` go
case "username":
	passwordAuthenticator.Username = value
	clusterConfig.Authenticator = passwordAuthenticator
case "password":
	passwordAuthenticator.Password = value
	clusterConfig.Authenticator = passwordAuthenticator
```

and do this

``` go
case "certPath":
	sslOpts.CertPath = value
	clusterConfig.SslOpts = &sslOpts
case "keyPath":
	sslOpts.KeyPath = value
	clusterConfig.SslOpts = &sslOpts
case "caPath":
	sslOpts.CaPath = value
	clusterConfig.SslOpts = &sslOpts
```